ngraph/core/reference/include/ngraph/runtime/reference/region_yolo.hpp

```diff
@@ -0,0 +1,179 @@
+                             const std::vector<int64_t>& mask,
+                             const int axis,
+                             const int end_axis)
+            {
+                NGRAPH_CHECK(input_shape.size() == 4);
+
+                const int batches = input_shape[0];
+                const int channels = input_shape[1];
+                const int height = input_shape[2];
+                const int width = input_shape[3];
+
+                const auto mask_size = mask.size();
+
+                std::copy(input, input + shape_size(input_shape), output);
+
+                int num_regions = 0;
+                int end_index = 0;
+
+                if (do_softmax)
+                {
+                    // Region layer (Yolo v2)
+                    num_regions = regions;
+                    end_index = width * height;
+                }
+                else
+                {
+                    // Yolo layer (Yolo v3)
+                    num_regions = mask_size;
+                    end_index = width * height * (classes + 1);
+                }
+
+                const int inputs_size = width * height * num_regions * (classes + coords + 1);
+
+                for (unsigned int batch_idx = 0; batch_idx < batches; batch_idx++)
+                {
+                    for (unsigned int n = 0; n < num_regions; n++)
+                    {
+                        int index = entry_index(width,
+                                                height,
+                                                coords,
+                                                classes,
+                                                inputs_size,
+                                                batch_idx,
+                                                n * width * height,
+                                                0);
+                        for (unsigned int i = index; i < index + 2 * width * height; i++)
+                        {
+                            output[i] = sigmoid<T>(output[i]);
+                        }
+
+                        index = entry_index(width,
+                                            height,
+                                            coords,
+                                            classes,
+                                            inputs_size,
+                                            batch_idx,
+                                            n * width * height,
+                                            coords);
+                        for (unsigned int i = index; i < index + end_index; i++)
+                        {
+                            output[i] = sigmoid<T>(output[i]);
+                        }
+                    }
+                }
+
+                if (do_softmax)
+                {
+                    int index =
+                        entry_index(width, height, coords, classes, inputs_size, 0, 0, coords + 1);
+                    int batch_offset = inputs_size / regions;
```

Comment:
Can we get a situation where remainder of the division is not 0?

Am I right that we always use lower bound for such cases?

Reply:
As far as I understand, yes we floor to the lower integer.